### ISSUE TITLE
S3 GetObject "SaveAs"

### ISSUE THREAD

**Issue by kennaji** [async-aws/aws#1205]
Hello,
the S3 GetObject "SaveAs" feature does not seem to work, I maybe have missed something but I cant see any way to batch copy a list of files from S3 to local while keeping their original names without using a file_put_contents for each one of them, which make the process much slower than when using PHP SDK CommandPool.

**Comment by jderusse** [comment]
Indeed, there are no such "SaveAs" properties in the official AWS API documentation. 
This is an helper provided by the PHP SDK.

I let the issue open if someone is will to contribute by adding a new `save` method in the `SimpleS3Client`
